app/src/main/java/com/android/wildex/ui/social/FriendScreenViewModel.kt

```diff
@@ -0,0 +1,351 @@
+ * @property isPending true if the current user has sent a friend request to this friend, false
+ *   otherwise
+ * @property isCurrentUser true if the friend is the current user, in which case no interactable
+ *   element should be displayed
+ */
+data class FriendState(
+    val friend: SimpleUser,
+    val isFriend: Boolean,
+    val isPending: Boolean,
+    val isCurrentUser: Boolean
+)
+
+/**
+ * ViewModel for the Friend Screen
+ *
+ * @param currentUserId the user id of the currently logged in user
+ * @param userRepository Repository used to get Simple users from
+ * @param userFriendsRepository Repository to get users' friends lists from
+ * @param friendRequestRepository Repository to get ongoing friend requests from
+ * @param userRecommender Recommendation algorithm to get suggested users from
+ */
+class FriendScreenViewModel(
+    private val currentUserId: Id = Firebase.auth.uid ?: "",
+    private val userRepository: UserRepository = RepositoryProvider.userRepository,
+    private val userFriendsRepository: UserFriendsRepository,
+    private val friendRequestRepository: RelationshipRepository =
+        RepositoryProvider.relationshipRepository,
+    private val userRecommender: UserRecommender =
+        UserRecommender(
+            currentUserId = currentUserId, userFriendsRepository = userFriendsRepository)
+) : ViewModel() {
+  /** Backing property for the friends screen state. */
+  private val _uiState = MutableStateFlow(FriendsScreenUIState())
+
+  /** Public immutable state exposed to the UI layer. */
+  val uiState: StateFlow<FriendsScreenUIState> = _uiState.asStateFlow()
+
+  private val suggestions = MutableStateFlow(listOf<RecommendationResult>())
+
+  /**
+   * Updates the UI state of the Friend Screen by fetching all needed data from the repositories.
+   *
+   * @param userId the user whose friend list we want to fetch
+   */
+  private suspend fun updateUIState(userId: Id) {
+    try {
+      val isCurrentUser = userId == currentUserId
+      val userFriends = userFriendsRepository.getAllFriendsOfUser(userId)
+      val currentUserFriends = userFriendsRepository.getAllFriendsOfUser(currentUserId)
+      val currentUserSentRequests =
+          friendRequestRepository.getAllPendingRelationshipsBySender(currentUserId)
+      val friendsStates =
+          userFriends.map {
+            FriendState(
+                friend = userRepository.getSimpleUser(it),
+                isFriend = currentUserFriends.contains(it),
+                isPending = currentUserSentRequests.any { request -> request.receiverId == it },
+                isCurrentUser = it == currentUserId)
+          }
+      suggestions.value = if (isCurrentUser) userRecommender.getRecommendedUsers() else emptyList()
+      val receivedRequests =
+          if (isCurrentUser) {
+            friendRequestRepository.getAllPendingRelationshipsByReceiver(currentUserId)
```

Comment:
Use the future new *getAllFriendRequestsByReceiver(receiverId: Id)*
```suggestion
            friendRequestRepository.getAllFriendRequestsByReceiver(currentUserId)
```